### PR TITLE
feat: add formulas 8.21 to 8.24 from FprEN 1992-1-1:2023 clause 8.2.1

### PR DESCRIPTION
**Pull request by egarciamendez**
## Description

Implements Formulas (8.21) to (8.24) of clause 8.2.1(5), for planar members with out-of-plane shear forces acting in two directions:

- `Form8Dot21DesignShearForcePerUnitWidth` for (8.21), the resultant design shear force per unit width.
- `Form8Dot22To24EffectiveDepth` for (8.22), (8.23) and (8.24), the effective depth as a function of the ratio of the shear forces. The three numbered formulas are branches of a single rule, so they share one class, following the pattern of `Form6Dot58And59TensileForce`. All three rows in the documentation table point at it.

Points worth a reviewer's attention:

- The branch boundaries in (8.22) to (8.24) are one-sided as printed. A ratio of exactly 0.5 falls under (8.22) and a ratio of exactly 2 falls under (8.24). Both have their own test case.
- The standard gives no rule for `v_Ed,x = 0`, where the ratio is undefined, and none for negative shear forces, where a negative ratio would land in (8.22) unintentionally. Both are rejected by validation and the choice is stated in the docstring.
- The two shear force components are taken as magnitudes in (8.21) as well, so the API is consistent across the clause.

The package rename of #1082 is merged, and this branch is rebased onto `main`. It does not depend on #1084.

Contributes to #1083

## Formulas as implemented

**Formula (8.21)**, `Form8Dot21DesignShearForcePerUnitWidth`

`equation`

$$
v_{Ed} = \sqrt{\left(v_{Ed,x}\right)^2 + \left(v_{Ed,y}\right)^2}
$$

`complete`

$$
v_{Ed} = \sqrt{\left(v_{Ed,x}\right)^2 + \left(v_{Ed,y}\right)^2} = \sqrt{\left(100.000\right)^2 + \left(40.000\right)^2} = 107.703 \ N/mm
$$

`complete_with_units`

$$
v_{Ed} = \sqrt{\left(v_{Ed,x}\right)^2 + \left(v_{Ed,y}\right)^2} = \sqrt{\left(100.000 \ N/mm\right)^2 + \left(40.000 \ N/mm\right)^2} = 107.703 \ N/mm
$$

**Formula (8.22/8.23/8.24)**, `Form8Dot22To24EffectiveDepth`

`equation`

$$
d = \begin{cases} d_x & \text{if } \frac{v_{Ed,y}}{v_{Ed,x}} \leq 0.5 \\ 0.5 \cdot (d_x + d_y) & \text{if } 0.5 < \frac{v_{Ed,y}}{v_{Ed,x}} < 2 \\ d_y & \text{if } \frac{v_{Ed,y}}{v_{Ed,x}} \geq 2 \end{cases}
$$

`complete`

$$
d = \begin{cases} d_x & \text{if } \frac{v_{Ed,y}}{v_{Ed,x}} \leq 0.5 \\ 0.5 \cdot (d_x + d_y) & \text{if } 0.5 < \frac{v_{Ed,y}}{v_{Ed,x}} < 2 \\ d_y & \text{if } \frac{v_{Ed,y}}{v_{Ed,x}} \geq 2 \end{cases} = \begin{cases} 300.000 & \text{if } \frac{100.000}{100.000} \leq 0.5 \\ 0.5 \cdot (300.000 + 250.000) & \text{if } 0.5 < \frac{100.000}{100.000} < 2 \\ 250.000 & \text{if } \frac{100.000}{100.000} \geq 2 \end{cases} = 275.000 \ mm
$$

`complete_with_units`

$$
d = \begin{cases} d_x & \text{if } \frac{v_{Ed,y}}{v_{Ed,x}} \leq 0.5 \\ 0.5 \cdot (d_x + d_y) & \text{if } 0.5 < \frac{v_{Ed,y}}{v_{Ed,x}} < 2 \\ d_y & \text{if } \frac{v_{Ed,y}}{v_{Ed,x}} \geq 2 \end{cases} = \begin{cases} 300.000 \ mm & \text{if } \frac{100.000 \ N/mm}{100.000 \ N/mm} \leq 0.5 \\ 0.5 \cdot (300.000 \ mm + 250.000 \ mm) & \text{if } 0.5 < \frac{100.000 \ N/mm}{100.000 \ N/mm} < 2 \\ 250.000 \ mm & \text{if } \frac{100.000 \ N/mm}{100.000 \ N/mm} \geq 2 \end{cases} = 275.000 \ mm
$$

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] Tests for both classes, all three branches, both boundaries and every raise
- [x] Docstrings added, with variable descriptions taken from the standard
- [x] Documentation table updated with the class names
- [x] `blueprints check` passes: lint, format, type check, and 100% coverage

## Summary by CodeRabbit

* **New Features**
  * Added Eurocode 2:2023 calculations for design shear force per unit width (Formula 8.21).
  * Added effective-depth calculations for planar members (Formulas 8.22–8.24), including ratio-based cases and input validation.
  * Added LaTeX representations for symbolic and numeric formula results.

* **Documentation**
  * Marked Formulas 8.21–8.24 as completed in the Eurocode documentation.

* **Tests**
  * Added coverage for calculations, boundary conditions, validation errors, and LaTeX output.

### CODE REVIEW

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_21.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            to the y direction [$N/mm$].
+        """
+        super().__init__()
+        self.v_ed_x = v_ed_x
+        self.v_ed_y = v_ed_y
+
+    @staticmethod
+    def _evaluate(v_ed_x: N_MM, v_ed_y: N_MM) -> N_MM:
+        """Evaluates the formula, for more information see the __init__ method."""
+        raise_if_negative(v_ed_x=v_ed_x, v_ed_y=v_ed_y)
```
Comment (rhhk-dot) [inline blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_21.py]: `v_ed_x` and `v_ed_y` should be able to be negative.

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_21.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        r"""[$v_{Ed}$] Design shear force per unit width in planar members [$N/mm$].
+
+        FprEN 1992-1-1:2023 (E) art 8.2.1 (5) - Formula (8.21)
+
+        Both components are taken as magnitudes, consistent with Formulas (8.22) to (8.24), which use the same
+        two quantities in a ratio that the standard prints without absolute-value bars.
+
+        Parameters
+        ----------
+        v_ed_x : N_MM
```
Comment (rhhk-dot) [inline blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_21.py:30]: I don't like the per unit width approach and would prefer N_mm2. However this is as it says in the Eurocode.

File: tests/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_21.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+from blueprints.validations import NegativeValueError
+
+
+class TestForm8Dot21DesignShearForcePerUnitWidth:
+    """Validation for formula 8.21 from FprEN 1992-1-1:2023."""
+
+    def test_evaluation(self) -> None:
+        """Tests the evaluation of the result."""
+        # Example values
+        v_ed_x = 100.0
```
Comment (rhhk-dot) [inline tests/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_21.py:14]: Although not wrong, preferbly use values that are 'realistic'

File: tests/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_21.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+        # Expected result, manually calculated
+        manually_calculated_result = 250.0  # N/mm
+
+        assert formula == pytest.approx(expected=manually_calculated_result, rel=1e-4)
+
+    @pytest.mark.parametrize(
+        ("v_ed_x", "v_ed_y"),
+        [
+            (-100.0, 40.0),  # v_ed_x is negative
```
Comment (rhhk-dot) [inline tests/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_21.py:39]: Negative values should be okay.

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_22_23_24.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        super().__init__()
+        self.v_ed_x = v_ed_x
+        self.v_ed_y = v_ed_y
+        self.d_x = d_x
+        self.d_y = d_y
+
+    @staticmethod
+    def _evaluate(v_ed_x: N_MM, v_ed_y: N_MM, d_x: MM, d_y: MM) -> MM:
+        """Evaluates the formula, for more information see the __init__ method."""
+        raise_if_negative(v_ed_y=v_ed_y, d_x=d_x, d_y=d_y)
```
Comment (rhhk-dot) [inline blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_22_23_24.py]: Working with absolute values for `v_ed_x` and `v_ed_y` fits better with formula (8.21)

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_22_23_24.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        raise_if_less_or_equal_to_zero(v_ed_x=v_ed_x)
```
Comment (rhhk-dot) [inline blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_22_23_24.py]: See comment line R51

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_22_23_24.py (hunk 5 above)
Comment (rhhk-dot) [reply blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_22_23_24.py]: `d_x` and `d_y` should be greater than 0

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_22_23_24.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+        ratio = v_ed_y / v_ed_x
+        if ratio <= 0.5:
+            return d_x
+        if ratio < 2:
+            return 0.5 * (d_x + d_y)
+        return d_y
+
+    def latex(self, n: int = 3) -> LatexFormula:
```
Comment (rhhk-dot) [inline blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_22_23_24.py:79]: Adding an intermediate result of 0.5 * (d_x + d_y) would be good

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_21.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+from blueprints.validations import raise_if_negative
+
+
+class Form8Dot21DesignShearForcePerUnitWidth(Formula):
+    """Class representing formula 8.21 for the calculation of the design shear force per unit width in planar members
+    with out-of-plane shear forces acting in two directions.
+    """
+
+    label = "8.21"
+    source_document = FPR_EN_1992_1_1_2023
```
Comment (rhhk-dot) [inline blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_21.py:17]: Check if naming is consistent with ec_gen2 branch

**Review by rhhk-dot — CHANGES_REQUESTED** [review]
Checked PR
